src/main/java/com/gluonhq/Main.java

```diff
@@ -91,14 +91,19 @@ public Double fromString(String s) {
         CheckBox editableProp = new CheckBox("Editable");
         editableProp.selectedProperty().bindBidirectional(editor.editableProperty());
 
+        Button undoButton = actionButton(LineAwesomeSolid.UNDO, editor.getActionFactory().undo());
+        undoButton.disableProperty().bind(editor.undoStackEmptyProperty());
```

Comment:
Not for this PR... but we should not expose this here. We should have `enabled` property on the `Action` which we can then bind to control in the demo. That property would be internally controlled by the state of the control. For example, hypothetically, if no selection,  text decoration actions will be disabled, disabling related controls.

Reply:
Sounds good.
So far the interface `Action` requires the viewModel as an argument for the `apply` method, but for the disabled/enabled property it doesn't seem  the right approach? Would an abstract class be more practical, given that we could create each action with the viewModel as a constructor argument, that  can be later on used by all the action methods?

Reply:
Very much possible... I was basically trying to create actions without state, i.e. they learn about viewModel only at execution time.  Action properties would require state, so passing viewModel would probably be the only choice. 
The downside of this is that all actions would have to be recreated, if viewModel changes. 
We could probably avoid that if, instead of passing model into it, we bind them somehow. I worry about that simply because, in my view, the model(text) will have to change, since control has to have that functionality.

Reply:
I'll file an issue for this so it can be done in a follow-up PR

Reply:
I'm not sure view model has to change if we redesign it a little. Currently it holds the text buffer. But if we can have text buffer as a property, then we can reassign it without  changing the viewModel itself. ViewModel will react on text buffer changing, triggering changes in the view.
So... changing a "document" on control will in essence change the text buffer bound to viewModel.

Reply:
Let's move this conversation to either #59 or #60